### PR TITLE
Python scripts for processing ROS odometry data into a CSV file with experiments annotated.

### PR DESCRIPTION
**Pull request by mshalm**
Two scripts for processing motion data captured as ROS odom messages stored in a rosbag:

`apriltag_csv.py` is a python script that converts the .bag format data into CSV time series data directly without post-processing.
`process_dynamics.py` is a python script that converts a .bag into full-state data for block tossing experiments by
1. Running `apriltag_csv.py`;
2. Using scipy interpolation tools to differentiate the raw pose data;
3. Using scipy signal tools to low-pass filter the resulting velocity data;
4. Detecting + labeling the starting & ending timestamps for the various tosses contained in the rosbag.

Both scripts were tested on real tossing data.

### CODE REVIEW

File: tools/apriltag/process_dynamics.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# true if block moves significantly over horizon H
+MOVING_CONDITION = lambda data, i: minBallOverHorizon(data[i:,:], TOSS_BALL_H) > TOSS_BALL_TOLERANCE
+
+# false if block moves significantly over horizon H
+STOPPED_CONDITION = lambda data, i: not MOVING_CONDITION(data, i)
+
+# returns minium ||\delta x|\ over a horizon of H in data
+def minBallOverHorizon(data, H):
+    diff_data = data[1:H,:] - data[0:(H-1),:]
+    return np.max(np.linalg.norm(diff_data,axis=1))
```
Comment (spfrommer) [inline tools/apriltag/process_dynamics.py]: Slightly confused why this returns the minimum Dx over a span of data. Shouldn't the return line read np.min?

Comment (spfrommer) [reply tools/apriltag/process_dynamics.py]: Alternatively, I feel like you might actually want the max movement based on the MOVING_CONDITION above, in which case maybe the function name should be adjusted

File: tools/apriltag/process_dynamics.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+# find first index of data for which condition() returns true
+def get_first_true(data,condition,H):
+    i = 0
+    while i<(data.shape[0]-(H-1)):
+        if condition(data, i):
+            return i
+        i = i+1
```
Comment (spfrommer) [inline tools/apriltag/process_dynamics.py]: More pythonic would be:
```
for i in range(0, data.shape[0] - (H-1))
```

File: tools/apriltag/process_dynamics.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        return (-1,0)
+    end_time = stopped_time + TOSS_BALL_H
+    return (start_time,end_time)
+
+# permutation of get_first_experiment to start at index start
+def get_first_experiment_after(start, data):
+    (s,e) = get_first_experiment(data[start:,:])
+    s = s + start
+    e = e + start
+    return (s,e)
```
Comment (spfrommer) [inline tools/apriltag/process_dynamics.py]: Could probably just combine these into return(s+start, e+start)

File: tools/apriltag/process_dynamics.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+print('python2 apriltag_csv.py ' + inbag + ' ' + csvfn)
+os.system('python2 apriltag_csv.py ' + inbag + ' ' + csvfn)
+outcsv = sys.argv[2]
+
+# get number of datapoints
+num_t = 0
+with open(csvfn,'r') as csvfile:
+    plots = csv.reader(csvfile, delimiter=',')
+    for row in plots:
+        num_t = num_t + 1
```
Comment (spfrommer) [inline tools/apriltag/process_dynamics.py]:
```
num_t += len(plots)
```

File: tools/apriltag/process_dynamics.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+with open(csvfn,'r') as csvfile:
+    plots = csv.reader(csvfile, delimiter=',')
+    for row in plots:
+        row = np.asarray([float(e) for e in row])
+        t[i] = row[0]
+        q_t[:,i] = row[1:5]
+        p_t[:,i] = row[5:8]
+        bq_t[:,i] = row[8:12]
+        bp_t[:,i] = row[12:15]
+        i = i + 1
```
Comment (spfrommer) [inline tools/apriltag/process_dynamics.py]: Probably could also use
```
for i, row in enumerate(plots)
```

File: tools/apriltag/process_dynamics.py (hunk 1 above)
Comment (mshalm) [reply tools/apriltag/process_dynamics.py]: renamed to maxDeltaOverHorizon

**Pull request merged**